extensions/typescript-language-features/src/tsServer/spawner.ts

```diff
@@ -85,10 +85,13 @@ export class TypeScriptServerSpawner {
 			this._tracer);
 	}
 
-	private getForkOptions(kind: ServerKind) {
+	private getForkOptions(kind: ServerKind, configuration: TypeScriptServiceConfiguration) {
 		const debugPort = TypeScriptServerSpawner.getDebugPort(kind);
 		const tsServerForkOptions: electron.ForkOptions = {
-			execArgv: debugPort ? [`--inspect=${debugPort}`] : [],
+			execArgv: [
+				...(debugPort ? [`--inspect=${debugPort}`] : []),
+				...(configuration.maxTsServerMemory ? [`--max-old-space-size=${configuration.maxTsServerMemory}`] : [])
```

Comment:
I think this has already validated to be an int (or, at least, a number), but I thought I should confirm.  We have some reason to believe we're not putting a raw string into the command line?

Reply:
I added additional validation to `readMaxTsServerMemory` to ensure the value in configuration is a non-negative safe integer.